// @forge/bridge/out/view/adf-renderer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createAdfRendererIframeProps = void 0;
const uuid_1 = require("uuid");
const createAdfRendererIframeProps = async (context, iframeId) => {
    const IframeResizer = await import('iframe-resizer');
    const IframeResizerModule = IframeResizer.default || IframeResizer;
    const origin = new URL(document.referrer).origin;
    const src = `${origin}/forge-apps/adf-renderer`;
    const id = iframeId || `forge-adf-renderer-iframe-${(0, uuid_1.v4)()}`;
    const onLoad = () => {
        var _a, _b;
        const iframe = document.getElementById(id);
        const message = {
            type: 'adf-document',
            document: (_a = context.extension.macro) === null || _a === void 0 ? void 0 : _a.body,
            timestamp: Date.now(),
            source: 'forge-adf-renderer',
            localId: context.localId
        };
        IframeResizerModule.iframeResizer({
            heightCalculationMethod: 'taggedElement',
            widthCalculationMethod: 'bodyScroll',
            initCallback: (iframe) => {
                var _a;
                (_a = iframe === null || iframe === void 0 ? void 0 : iframe.iFrameResizer) === null || _a === void 0 ? void 0 : _a.resize();
            }
        }, iframe || '');
        (_b = iframe === null || iframe === void 0 ? void 0 : iframe.contentWindow) === null || _b === void 0 ? void 0 : _b.postMessage(message, origin);
    };
    setTimeout(() => {
        document.documentElement.style.height = 'auto';
        document.body.style.height = 'auto';
    }, 200);
    return {
        id,
        src,
        onLoad
    };
};
exports.createAdfRendererIframeProps = createAdfRendererIframeProps;
